src/components/codeblocks/CodeblockMermaid.svelte

```diff
@@ -14,10 +14,17 @@
 	import { Paths } from "src/utils/paths";
 	import { wrap_in_codeblock } from "src/utils/strings";
 	import CodeblockErrors from "./CodeblockErrors.svelte";
+	import { onMount } from "svelte";
 
 	export let plugin: BreadcrumbsPlugin;
 	export let options: ICodeblock["Options"];
 	export let errors: BreadcrumbsError[];
+	export let file_path: string;
+
+	let all_paths = [];
+
+	// if the file_path is an empty string, so the code block is not rendered inside note, we fall back to the active file store
+	$: active_file_path = file_path ? file_path : ($active_file_store ? $active_file_store.path : undefined);
```

Comment:
We should fallback to `''` instead of `undefined`, right? Most callers expect start_node to atleast be an empty string

Reply:
And if we just fallback to `''`, then the `file_path` prop is always fine, we don't even need to check `active_file_store`

Reply:
Either way, `get_all_paths` will just return `[]` in the else case, but ensuring a string makes TS happy